core/src/main/java/org/elasticsearch/index/mapper/core/KeywordFieldMapper.java

```diff
@@ -128,6 +120,8 @@ public KeywordFieldMapper build(BuilderContext context) {
                 } else if (propName.equals("ignore_above")) {
                     builder.ignoreAbove(XContentMapValues.nodeIntegerValue(propNode, -1));
                     iterator.remove();
+                } else if (TypeParsers.parseNorms(builder, propName, propNode, parserContext)) {
```

Comment:
Wouldn't the definition of "upgradable" for string to text/keyword mean the norms setting fits with what is allowed? As this is now, it would mean eg keyword fields could have all of the old settings right, but they would be deprecated...that is just really weird for a new mapper type.